flink-java/src/main/java/org/apache/flink/api/java/functions/SelectByMinFunction.java

```diff
@@ -72,8 +73,8 @@ public T reduce(T value1, T value2) throws Exception {
 		for (int position : fields) {
 			// Save position of compared key
 			// Get both values - both implement comparable
-			Comparable comparable1 = value1.getFieldNotNull(position);
-			Comparable comparable2 = value2.getFieldNotNull(position);
+			Comparable comparable1 = ((Tuple)value1).getFieldNotNull(position);
```

Comment:
The `CaseClassTypeInfo` is the type information which is created for Scala tuples, if I'm not mistaken. And all Scala tuples are of type `Product`. With that you should be able to adapt the `SelectByMax/MinFunction`.